libs/geo/src/main/java/org/elasticsearch/geo/utils/GeographyValidator.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.geo.utils;
+
+import org.elasticsearch.geo.geometry.Circle;
+import org.elasticsearch.geo.geometry.Geometry;
+import org.elasticsearch.geo.geometry.GeometryCollection;
+import org.elasticsearch.geo.geometry.GeometryVisitor;
+import org.elasticsearch.geo.geometry.Line;
+import org.elasticsearch.geo.geometry.LinearRing;
+import org.elasticsearch.geo.geometry.MultiLine;
+import org.elasticsearch.geo.geometry.MultiPoint;
+import org.elasticsearch.geo.geometry.MultiPolygon;
+import org.elasticsearch.geo.geometry.Point;
+import org.elasticsearch.geo.geometry.Polygon;
+import org.elasticsearch.geo.geometry.Rectangle;
+
+/**
+ * Validator that checks that lats are between -90 and +90 and lons are between -180 and +180 and altitude is present only if
+ * ignoreZValue is set to true
+ */
+public class GeographyValidator implements GeometryValidator {
```

Comment:
I ran into this in #40869  Because we don't include lucene dependencies in our library plugins I had to copy much of Lucene's `BitUtils` and `GeoHashUtils` implementation. Here we're having to copy some of Lucene's `GeoUtils` functionality.  Starting to get a little concerned w/ the code duplication and am curious as to whether or not it makes sense to start allowing lucene dependencies in our Libraries? Will this break the build configuration or just our library philosophy?  /cc @rjernst @atorok

Reply:
If we want to reuse lucene pieces we would have to jump through a lot of hoops and probably move validation to server, since we need libs/geo in SQL client and we really don't want it to depend on it. I don't think it is worth it because of 2 constants.

Reply:
Yeah. I'm not so concerned about the two constants here as I am about the growing duplication of code, like `IOUtils`, `BitUtils`, `GeoHashUtils`, and now `GeoUtils`. Perhaps the concern is premature, but I'm starting to see this become a bit of a trend. I just don't want to wake up one day and see that it got out of hand and future maintenance becomes a nightmare. It doesn't have to hold up this PR but I think it does warrant an async discussion.

Reply:
Worst case, in a distant future if it gets out of hand, we could have a compile time shading/copying of relevant lucene classes, if we want to include these in our lib without adding an external jar dep. I agree with Igor we should not add deps to SQL client.

Reply:
Awesome... thx @rjernst

Reply:
Would it be conceivable to have something like `lucene-utils` produced upstream on which we _could_ depend on ? That could be a long term solution too, would be the same as @rjernst suggested, but would allow us to add a clean dependency.

Reply:
I don't think this would be likely, since from Lucene's perspective there is no reason for this extra dependency. Lucene doesn't want to be a general purpose utility library.